apps/avatax/scripts/cleanup-client-logs.ts

```diff
@@ -0,0 +1,101 @@
+import { parseArgs } from "node:util";
+
+import { DeleteItemCommand, QueryCommand } from "dynamodb-toolbox";
+import { saleorApp } from "saleor-app";
+
+import { env } from "@/env";
+import {
+  createLogsDocumentClient,
+  createLogsDynamoClient,
+} from "@/modules/client-logs/dynamo-client";
+import { ClientLogDynamoEntityFactory, LogsTable } from "@/modules/client-logs/dynamo-schema";
+
+const { values } = parseArgs({
+  options: {
+    "dry-run": { type: "boolean", short: "d" },
+  },
+});
+
+const today = new Date();
+const endDate = new Date(today.getTime() - 14 * 24 * 60 * 60 * 1000); // 14 days from today
+
+const main = async () => {
+  const start = performance.now();
+
+  const dynamoClient = createLogsDynamoClient();
+
+  const logsTable = LogsTable.create({
+    documentClient: createLogsDocumentClient(dynamoClient),
+    tableName: env.DYNAMODB_LOGS_TABLE_NAME,
+  });
+
+  const logsByCheckoutOrOrderId =
+    ClientLogDynamoEntityFactory.createLogByCheckoutOrOrderId(logsTable);
+  const logsByDateEntity = ClientLogDynamoEntityFactory.createLogByDate(logsTable);
+
+  const appInstallations = await saleorApp.apl.getAll().catch(() => {
+    console.error("Could not fetch instances from the APL");
+
+    process.exit(1);
+  });
+
+  for (const { saleorApiUrl, appId } of appInstallations) {
+    try {
+      let lastEvaluatedKey: Record<string, unknown> | undefined = undefined;
+
+      const command = logsTable
+        .build(QueryCommand)
+        .query({
+          partition: LogsTable.getPrimaryKey({ saleorApiUrl, appId }),
+          lte: endDate,
+        })
+        .entities(logsByCheckoutOrOrderId, logsByDateEntity);
+
+      if (values["dry-run"]) {
+        console.log(`Would delete logs for ${saleorApiUrl}#${appId}`);
```

Comment:
Maybe log also timestamp of first & last in the dry run so you dont accidentaly target wrong query

Reply:
Added in https://github.com/saleor/apps/pull/1762/commits/fa9fa15e3c833c35ca3712884a5673a21183f45f